src/types.ts

```diff
@@ -54,41 +52,26 @@ export type RequestPayloadOf<
 
 type RestrictedUseQueryOptions<
   Response,
-  TError = unknown,
+  TError = DefaultError,
   Data = Response,
 > = Omit<UseQueryOptions<Response, TError, Data>, 'queryKey' | 'queryFn'> & {
   axios?: AxiosRequestConfig;
 };
 
-type RestrictedUseInfiniteQueryOptions<
-  Response,
-  Request,
-  Data = Response,
-> = Omit<
-  UseInfiniteQueryOptions<Response, unknown, Data>,
-  'queryKey' | 'queryFn' | 'getNextPageParam' | 'getPreviousPageParam'
+type RestrictedUseInfiniteQueryOptions<Response, Request> = Omit<
+  UseInfiniteQueryOptions<InfiniteData<Response>, DefaultError>,
+  | 'queryKey'
+  | 'queryFn'
+  | 'initialPageParam'
+  | 'getNextPageParam'
+  | 'getPreviousPageParam'
 > & {
   axios?: AxiosRequestConfig;
-  getNextPageParam?: (lastPage: Response) => Partial<Request> | undefined;
+  initialPageParam: Partial<Request>; // use init payload?
```

Comment:
I don't think we should use the initial payload. I'd suggest removing the comment -- the current state seems right to me.

Reply:
ah missed the stray comment when I was wrapping up